docs/module_federation/nextjs_plugin.md

```diff
@@ -0,0 +1,252 @@
+---
+id: nextjs_plugin
+title: Stacks with NextJs Module Federation
+sidebar_label: Stacks Module Federation
+description: Using stacks with NextJs Module Federation
+keywords:
+  - Nx
+  - monorepo
+  - stacks
+  - Generator
+  - Executor
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+Microfrontends are an architectural pattern for building frontend applications by breaking them down into smaller, loosely coupled, and independently deployable parts called "micro-frontends." Each microfrontend corresponds to a distinct feature or functionality of the application and is developed and deployed independently of the others.
+
+#### Module Federation with NextJs
+
+Module Federation is a feature introduced in Webpack 5, it enables developers to share code dynamically across different applications at runtime. This feature is particularly beneficial when building microfrontends or decoupled frontend architectures.
+
+@module-federation/nextjs-mf is a plugin that leverages Webpack 5’s Module Federation features, allowing developers to implement Module Federation in NextJs applications.
+
+It aims to enable a federated modules architecture in NextJs, enabling you to split a monolithic NextJs application into smaller, independently deployable parts.
+
+The following guide details the steps of how to set up Module Federation with a Stacks NextJs application. You can also visit this [GitHub repo](https://github.com/amido/stacks-nextjs-federated-modules-example) to view an existing example application
+
+## NextJs Module Federation with Stacks
+
+### Prerequisites
+
+#### Nx
+
+We recommend installing Nx globally, all Nx based commands in this guide are based upon a globally installed Nx package.
```

Comment:
I am a believer that if you can have it local and not global then that is the preference for NPM packages and I feel that this can be misunderstood.

Maybe something like: `All Nx based commands in this guide are based upon a globally installed Nx package. This is how we recommend installing NX but our guides also work using npx (link to npx)`

What do you think Lee?

Reply:
Yes i agree with local as a preference when possible, I just used the same text around Nx as we have in our other docs here https://stacks.amido.com/docs/nx/nx_stacks#, for consistency.
Happy to update though.

Reply:
Ahhh ok, lets keep it as it is and look to update the wording when we work on the site later this month